### ISSUE TITLE
Writing to CSV results in partially corrupted output

### ISSUE THREAD

**Issue by hskrasek** [thephpleague/csv#148]
Currently running into an issue where writing data to CSVs in production is resulting in corrupted data on the second row, across multiple servers. The first row of data is correct, and the third row of data -Nth row of data is correct, but the second row of data always contains corrupted data.

When I open the file in Sublime Text I get this (sorry github doesn't like Dropbox screenshot embedding):
https://www.dropbox.com/s/ud0tbhyc7x5orkk/Screenshot%202016-01-14%2011.53.54.png?dl=0

And in VIM it just becomes a slew of this:
```
@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
```

Some character is being inserted that VIM and sublime interpret, TextEdit on Mac just strips the character. Trying to think of what else you might need, but can't think of anything else.

PHP: 5.5.30
Platform: CentOS 6.5
Package: 7.1.2

**Comment by nyamsprod** [comment]
Have you tried the latest version ? Since **2** new versions where released since 7.1.2

**Comment by hskrasek** [comment]
So an update here, I found out the code that was written to write the CSV data to file was not using an SplFileObject, but an SplTempFileObject and then writing this data to file in another manner. To get the contents of the Writer, the code was triggering it's __toString method which uses `ob_start` and `ob_get_clean`. The second we moved the code away from using the output buffer, our corruption issue was resolved.
